ui/app/selectors/confirm-transaction.js

```diff
@@ -133,7 +134,8 @@ export const tokenAmountAndToAddressSelector = createSelector(
       const toParam = params.find(param => param.name === TOKEN_PARAM_TO)
       const valueParam = params.find(param => param.name === TOKEN_PARAM_VALUE)
       toAddress = toParam ? toParam.value : params[0].value
-      tokenAmount = valueParam ? Number(valueParam.value) : Number(params[1].value)
+      const value = valueParam ? Number(valueParam.value) : Number(params[1].value)
+      tokenAmount = roundExponential(value)
```

Comment:
`roundExponential` seems out of place in selectors. It is an operation done just to modify the way the value is presented in the view. The calls of `roundExponential` in `confirm-token-transaction-base.component.js` seem to be in the right place.

I like the idea of keeping selectors (as much as possible) focused on "selecting": getting the needed data out of state. And then doing manipulation or transformation on data elsewhere.

Alternatively, I could easily be convinced to do it this way: treat selectors methods that are meant to both get data from state AND transform it to exactly what is needed by the view. In this scenario, there would be no operations on data in components. A component might decide which of two variables to render, but would render the variable as it was received.

So, as long we consistently do it one way or the other, I am good.

Now, what I've described above is likely inconsistent with a good chunk of the codebase. If the code in this PR can easily be refactored to the above changes, then we can do it now. Otherwise, if it is more complicated, we can merge this PR as is and make an issue for a larger refactor along these lines.

Reply:
That can be a good approach. I've been thinking of selectors as methods to get usable data from state, ie your alternate proposal. I don't think it's bad practice to modify data within selectors, as seen [here](https://github.com/reduxjs/reselect#api) in the `reselect` library docs, since it could lead to less duplication if you're always doing the same transformation wherever the selector is used.